Staale/docker images (#123)
LucidtechAI/cradl-docs: docs/get-started/creating-a-workflow.mdx

:::info
Auto-generated workflows are currently only supported by the CLI.
:::

### Workflows, transitions and executions
A workflow is defined by a series of _transitions_ which mutate the state of your _workflow execution_. There are two types of transitions; _manual_
transition which mutates the state based on input from a user, and _docker_ transition which mutates the state programatically in a Docker-container.
At each step in the workflow, the current state is given as input to the transition, and the output of the transition is the new state.
When a new workflow execution is created, an _initial state_ is provided.

## Starting a workflow execution
Let's test our new workflow by creating a new workflow execution. The initial state of the workflow executions is provided as a JSON object, and the workflow we just generated assumes that the initial state is on the form
`{"documentId": "las:document:<document-id>"}`.

<Tabs>
<TabItem value="bash" label="CLI">

```bash
$ las create document mydocument.pdf > input.json

$ las workflows execute las:workflow:<my-workflow-id> input.json
{
  "workflowId": "las:workflow:<my-workflow-id>"

>   "executionId": "las:workflow-executions:<my-execution-id>"

Reviewer: ```suggestion
  "executionId": "las:workflow-execution:<my-execution-id>"
```

  ...
}
```

</TabItem>
</Tabs>

Now that we have created a workflow execution, the following steps are executed:

1. The initial state will be provided as input to the first transition (_Preprocess_).
   The transition will create a [Prediction](/concepts/predictions) on the provided document.

> 2. If the _confidence_ of each field predicted is below certain thresholds, a manual transition will be invoked

Reviewer: ```suggestion
2. If the _confidence_ of any field predicted is below certain thresholds, a manual transition will be invoked
```

   so that an end user can validate that the predictions are correct.
3. In the last transition (_Postprocess), the _ground truth_ of the document is updated and the document is assigned to a [Dataset](/concepts/datasets)
   so that it can be used for training. This transition is also responsible for _exporting_ the document.

## Exporting documents

> To customize the export logic, we have two options:

Reviewer: Maybe this is clearer?

```suggestion
To customize the exporting functionality, we have two options:
```
